addons/fullscreen/userscript.js

```diff
@@ -31,6 +31,58 @@ export default async function ({ addon, console }) {
     }
   }
 
+  // The "phantom header" is a small strip at the top of the page that
+  // brings the header into view when hovered.
+  async function updatePhantomHeader() {
+    if (
+      !addon.self.disabled &&
+      addon.tab.redux.state.scratchGui.mode.isFullScreen &&
+      addon.settings.get("hideToolbar") &&
+      addon.settings.get("hoverToolbar")
+    ) {
+      const canvas = await addon.tab.waitForElement('[class*="stage_full-screen"] canvas');
+      const header = await addon.tab.waitForElement('[class^="stage-header_stage-header-wrapper"]');
+      const phantom = header.parentElement.appendChild(document.createElement("div"));
+      phantom.classList.add("phantom-header");
+
+      // Make the header a child of the phantom, so that mouseleave will trigger when the
+      // mouse leaves the header OR the phantom header.
+      phantom.appendChild(header);
+
+      phantom.addEventListener("mouseenter", () => {
+        header.classList.add("stage-header-hover");
+      });
+      phantom.addEventListener("mouseleave", () => {
+        header.classList.remove("stage-header-hover");
+      });
+
```

Comment:
Might as well add a comment for this:
```suggestion

      // Pass click events on the phantom header onto the project player, essentially making it click-through
```